src/replica/replica_stub.cpp

```diff
@@ -600,9 +646,30 @@ void replica_stub::initialize(const replication_options &opts, bool clear /* = f
         }
     }
 
+    std::string encryption_key;
+    std::string iv;
+    std::string key_version;
+    std::string server_key;
+    // get and store eek from kms
+    if (key_provider && !utils::is_empty(FLAGS_hadoop_kms_url)) {
+        get_kms_key(_options.data_dirs[0], &encryption_key, &iv, &key_version);
+        if (encryption_key.empty()) {
```

Comment:
Add comments to describe when it's empty and non-empty, and what will do in the both cases.